### ISSUE TITLE
In docs.yml, "Build C++ WASM" fails with `ninja` not found

### ISSUE THREAD

**Issue by jpivarski** [scikit-hep/awkward#3329]
For example, in this: https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054

The failing log is:

```
Run # pyodide-build doesn't work out of the box with pipx
  # pyodide-build doesn't work out of the box with pipx
  CFLAGS=-fexceptions LDFLAGS=-fexceptions pyodide build --exports whole_archive
  shell: /usr/bin/bash -e {0}
  env:
    X86_64_PYTHON_VERSION: 3.11.0
    SOURCE_DATE_EPOCH: 1668811[2](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:2)11
    pythonLocation: /opt/hostedtoolcache/Python/3.11.2/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/[3](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:3).11.2/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.2/x6[4](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:4)
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.2/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.2/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.2/x64/lib
    PATH: /home/runner/work/_temp/[5](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:5)5485d84-5b5a-4[6](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:6)3f-b5b9-a3a3e161e9e8/emsdk-main:/home/runner/work/_temp/55485d84-5b5a-463f-b5b9-a3a3e161e9e8/emsdk-main/upstream/emscripten:/opt/hostedtoolcache/Python/3.11.2/x64/bin:/opt/hostedtoolcache/Python/3.11.2/x64:/snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
    EMSDK: /home/runner/work/_temp/55485d84-5b5a-463f-b5b9-a3a3e161e9e8/emsdk-main
    EMSDK_NODE: /home/runner/work/_temp/55485d84-5b5a-463f-b5b9-a3a3e161e9e8/emsdk-main/node/20.18.0_64bit/bin/node
  
*** scikit-build-core 0.10.[7](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:7) using CMake 3.31.1 (wheel)
*** Configuring CMake...
configure: cmake -DCMAKE_C_COMPILER=/tmp/tmpkwce4mjl/cc -DCMAKE_CXX_COMPILER=/tmp/tmpkwce4mjl/c++ -DCMAKE_AR=/tmp/tmpkwce4mjl/ar -DCMAKE_C_COMPILER_AR=/tmp/tmpkwce4mjl/ar -DCMAKE_CXX_COMPILER_AR=/tmp/tmpkwce4mjl/ar --fresh -S. -Bbuild/cpython-311 -DCMAKE_BUILD_TYPE:STRING=Release -Cbuild/cpython-311/CMakeInit.txt -DCMAKE_INSTALL_PREFIX=/tmp/tmpn02_aikz/wheel/platlib -DCMAKE_MAKE_PROGRAM=ninja -DCMAKE_CROSSCOMPILING_EMULATOR=/home/runner/work/_temp/554[8](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:8)5d84-5b5a-463f-b5b9-a3a3e161e9e8/emsdk-main/node/20.18.0_64bit/bin/node
loading initial cache file build/cpython-311/CMakeInit.txt
CMake Error at CMakeLists.txt:6 (project):
  Running
   'ninja' '--version'
  failed with:
   no such file or directory
-- Configuring incomplete, errors occurred!
emcmake: error: 'cmake -DCMAKE_C_COMPILER=/tmp/tmpkwce4mjl/cc -DCMAKE_CXX_COMPILER=/tmp/tmpkwce4mjl/c++ -DCMAKE_AR=/tmp/tmpkwce4mjl/ar -DCMAKE_C_COMPILER_AR=/tmp/tmpkwce4mjl/ar -DCMAKE_CXX_COMPILER_AR=/tmp/tmpkwce4mjl/ar --fresh -S. -Bbuild/cpython-311 -DCMAKE_BUILD_TYPE:STRING=Release -Cbuild/cpython-311/CMakeInit.txt -DCMAKE_INSTALL_PREFIX=/tmp/tmpn02_aikz/wheel/platlib -DCMAKE_MAKE_PROGRAM=ninja -DCMAKE_CROSSCOMPILING_EMULATOR=/home/runner/work/_temp/55485d84-5b5a-463f-b5b[9](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:9)-a3a3e161e9e8/emsdk-main/node/20.18.0_64bit/bin/node' failed (returned 1)
*** CMake configuration failed
* Creating virtualenv isolated environment...
* Installing packages in isolated environment... (pybind11, scikit-build-core>=0.[10](https://github.com/scikit-hep/awkward/actions/runs/12188354714/job/34002123054#step:9:10))
* Getting dependencies for wheel...
* Installing packages in isolated environment... (ninja>=1.5)
* Building wheel...
ERROR Backend subproccess exited when trying to invoke build_wheel
Error: Process completed with exit code 1.
```

and it was configured somewhere in

https://github.com/scikit-hep/awkward/blob/661702bd9f448333bf48e9768cdc0c98603470c3/.github/workflows/docs.yml#L61-L118

I think it's failing specifically on

https://github.com/scikit-hep/awkward/blob/661702bd9f448333bf48e9768cdc0c98603470c3/.github/workflows/docs.yml#L111

so `pyodide build`. Is this a change in Pyodide? Does it now need Ninja to be installed, and would that be easy to add?

**Comment by agoose77** [comment]
@jpivarski Ninja should be added by scikit-build-core, unless it's not available on the platform.

However, this machinery was added so that we can deploy the latest version of Awkward Array in docs, with a JupyterLite distribution.

It looks like you removed that functionality, and I just loaded the `/lite` API path and found it doesn't work. So, if you're not interested in this functionality and shipping JupyterLite, then the quickest fix is just remove any logic pertaining to building of WASM wheels in CI!
